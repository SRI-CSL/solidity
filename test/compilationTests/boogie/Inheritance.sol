pragma solidity ^0.4.17;

contract Base {
    function f() public pure returns (uint) {
        return 1;
    }

    function g() public pure returns (uint) {
        return 2;
    }
}

contract Derived is Base {
    function f() public pure returns (uint) {
        return 3;
    }

    function h() public pure returns (uint) {
        return f() + g();
    }

    function __verifier_main() public pure {
        assert(h() == 5);
    }
}