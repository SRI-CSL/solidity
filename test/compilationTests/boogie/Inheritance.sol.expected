Verifying Base::f: OK
Verifying Base::g: OK
Verifying Inheritance::f: OK
Verifying Inheritance::h: OK
No errors found.
